examples/classification/plot_digits_classification.py

```diff
@@ -22,49 +19,84 @@
 from sklearn import datasets, svm, metrics
 from sklearn.model_selection import train_test_split
 
-# The digits dataset
+###############################################################################
+# Digits dataset
+# --------------
+#
+# The digits dataset consists of 8x8
+# pixel images of digits. The ``images`` attribute of the dataset stores
+# 8x8 arrays of grayscale values for each image. We will use these arrays to
+# visualize the first 4 images. The ``target`` attribute of the dataset stores
+# the digit each image represents and this is included in title of the 4 plots
+# below.
+#
+# Note: if we were working from image files (e.g., 'png' files), we could load
+# them using :func:`matplotlib.pyplot.imread`.
+
 digits = datasets.load_digits()
 
-# The data that we are interested in is made of 8x8 images of digits, let's
-# have a look at the first 4 images, stored in the `images` attribute of the
-# dataset.  If we were working from image files, we could load them using
-# matplotlib.pyplot.imread.  Note that each image must have the same size. For these
-# images, we know which digit they represent: it is given in the 'target' of
-# the dataset.
-_, axes = plt.subplots(2, 4)
+_, axes = plt.subplots(1, 4, figsize=(10, 3))
 images_and_labels = list(zip(digits.images, digits.target))
-for ax, (image, label) in zip(axes[0, :], images_and_labels[:4]):
+for ax, (image, label) in zip(axes, images_and_labels[:4]):
```

Comment:
```suggestion
for ax, image, label in zip(axes, digits.images, digits.target):
```

Since zip will loop on the smallest iterable, we don't need to have `[:4]`